### ISSUE TITLE
Docker fails to run image with error "Invalid Argument"

### ISSUE THREAD

**Issue by harche** [moby/moby#14904]
`docker.log`

```
INFO[0000] [graphdriver] using prior storage driver "aufs" 
WARN[0000] Running modprobe bridge nf_nat failed with message: , error: exit status 1 
WARN[0000] Your kernel does not support swap memory limit. 
INFO[0001] Loading containers: start.                   

INFO[0001] Loading containers: done.                    
INFO[0001] Daemon has completed initialization          
INFO[0001] Docker daemon                                 commit=786b29d execdriver=native-0.2 graphdriver=aufs version=1.7.1
INFO[0001] GET /v1.19/version                           
INFO[0011] POST /v1.19/containers/create                
ERRO[0011] Handler for POST /containers/create returned error: No such image: hello-world (tag: latest) 
ERRO[0011] HTTP Error                                    err=No such image: hello-world (tag: latest) statusCode=404
INFO[0011] POST /v1.19/images/create?fromImage=hello-world&tag=latest 
INFO[0018] Image manifest for hello-world:latest has been verified 
INFO[0020] POST /v1.19/containers/create                
INFO[0020] POST /v1.19/containers/86f358024cae4895a3b87282b9db0f390e29fb73b68dd3ab1b3345335a15c135/attach?stderr=1&stdout=1&stream=1 
INFO[0020] POST /v1.19/containers/86f358024cae4895a3b87282b9db0f390e29fb73b68dd3ab1b3345335a15c135/start 
ERRO[0020] leaving endpoint failed: a container has already joined the endpoint 
ERRO[0020] Handler for POST /containers/{name:.*}/start returned error: Cannot start container 86f358024cae4895a3b87282b9db0f390e29fb73b68dd3ab1b3345335a15c135: invalid argument 
ERRO[0020] HTTP Error                                    err=Cannot start container 86f358024cae4895a3b87282b9db0f390e29fb73b68dd3ab1b3345335a15c135: invalid argument statusCode=500
```

`docker info` 

```
Containers: 15
Images: 43
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 73
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-57-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 4
Total Memory: 7.558 GiB
Name: ThinkPad-T420
ID: ZH5Z:JQDT:QRKU:NUZX:ZNBN:X6EH:XWAN:P753:KRJ5:XQWR:AWFE:O2HU
WARNING: No swap limit support
```

`docker run hello-world` 

```
Error response from daemon: Cannot start container 6b4b0057d27b1507e1436c5bd3bc1e5cd3d50e8e2e556443719d64f1950da6a5: invalid argument
```

`docker version`

```
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```

`uname -a`

```
Linux harshal-ThinkPad-T420 3.13.0-57-generic #95-Ubuntu SMP Fri Jun 19 09:28:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

`/var/log/syslog`

```
Jul 23 21:29:03 harshal-ThinkPad-T420 acvpnagent[2309]: Function: tableCallbackHandler File: RouteMgr.cpp Line: 1723 Invoked Function: recv Return Code: 11 (0x0000000B) Description: unknown 
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.253011] device vetheafcc75 entered promiscuous mode
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.254015] IPv6: ADDRCONF(NETDEV_UP): veth9a2b6b2: link is not ready
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: Function: tableCallbackHandler File: RouteMgr.cpp Line: 1723 Invoked Function: recv Return Code: 11 (0x0000000B) Description: unknown 
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: A new network interface has been detected.
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: Function: logInterfaces File: RouteMgr.cpp Line: 2105 Invoked Function: logInterfaces Return Code: 0 (0x00000000) Description: IP Address Interface List: 192.168.0.104 9.84.216.231 FE80:0:0:0:8E70:5AFF:FE57:C6D0 FE80:0:0:0:42:ACFF:FE11:6  
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: Function: tableCallbackHandler File: RouteMgr.cpp Line: 1723 Invoked Function: recv Return Code: 11 (0x0000000B) Description: unknown 
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.255476] IPv6: ADDRCONF(NETDEV_CHANGE): veth9a2b6b2: link becomes ready
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: devices added (path: /sys/devices/virtual/net/veth9a2b6b2, iface: veth9a2b6b2)
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: device added (path: /sys/devices/virtual/net/veth9a2b6b2, iface: veth9a2b6b2): no ifupdown configuration found.
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]: <warn> /sys/devices/virtual/net/veth9a2b6b2: couldn't determine device driver; ignoring...
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: devices added (path: /sys/devices/virtual/net/vetheafcc75, iface: vetheafcc75)
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: device added (path: /sys/devices/virtual/net/vetheafcc75, iface: vetheafcc75): no ifupdown configuration found.
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]: <warn> /sys/devices/virtual/net/vetheafcc75: couldn't determine device driver; ignoring...
Jul 23 21:29:18 harshal-ThinkPad-T420 avahi-daemon[1112]: Withdrawing workstation service for veth9a2b6b2.
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: devices removed (path: /sys/devices/virtual/net/veth9a2b6b2, iface: veth9a2b6b2)
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.339159] docker0: port 1(vetheafcc75) entered disabled state
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.340884] device vetheafcc75 left promiscuous mode
Jul 23 21:29:18 harshal-ThinkPad-T420 kernel: [186663.340904] docker0: port 1(vetheafcc75) entered disabled state
Jul 23 21:29:18 harshal-ThinkPad-T420 avahi-daemon[1112]: Withdrawing workstation service for vetheafcc75.
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: message repeated 3 times: [ Function: tableCallbackHandler File: RouteMgr.cpp Line: 1723 Invoked Function: recv Return Code: 11 (0x0000000B) Description: unknown ]
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: A network interface has gone down.
Jul 23 21:29:18 harshal-ThinkPad-T420 acvpnagent[2309]: Function: logInterfaces File: RouteMgr.cpp Line: 2105 Invoked Function: logInterfaces Return Code: 0 (0x00000000) Description: IP Address Interface List: 192.168.0.104 9.84.216.231 FE80:0:0:0:8E70:5AFF:FE57:C6D0  
Jul 23 21:29:18 harshal-ThinkPad-T420 NetworkManager[1233]:    SCPlugin-Ifupdown: devices removed (path: /sys/devices/virtual/net/vetheafcc75, iface: vetheafcc75)
Jul 23 21:29:28 harshal-ThinkPad-T420 wpa_supplicant[1534]: wlan0: CTRL-EVENT-SCAN-STARTED 
Jul 23 21:29:32 harshal-ThinkPad-T420 wpa_supplicant[1534]: nl80211: send_and_recv->nl_recvmsgs failed: -33
```

**Comment by phemmer** [comment]
This line is the likely culprit:

    WARN[0000] Running modprobe bridge nf_nat failed with message: , error: exit status 1 

You appear to be missing kernel support. As it is, containers will likely only run with host network mode (`docker run --net=host`).

If you can run [this script](https://github.com/docker/docker/blob/master/contrib/check-config.sh), it will tell you exactly what is missing.

**Comment by harche** [comment]
` ./test.sh`
``` 
warning: /proc/config.gz does not exist, searching other paths for kernel config...
info: reading kernel config from /boot/config-3.13.0-57-generic ...

Generally Necessary:
- cgroup hierarchy: properly mounted [/sys/fs/cgroup]
- apparmor: enabled and tools installed
- CONFIG_NAMESPACES: enabled
- CONFIG_NET_NS: enabled
- CONFIG_PID_NS: enabled
- CONFIG_IPC_NS: enabled
- CONFIG_UTS_NS: enabled
- CONFIG_DEVPTS_MULTIPLE_INSTANCES: enabled
- CONFIG_CGROUPS: enabled
- CONFIG_CGROUP_CPUACCT: enabled
- CONFIG_CGROUP_DEVICE: enabled
- CONFIG_CGROUP_FREEZER: enabled
- CONFIG_CGROUP_SCHED: enabled
- CONFIG_CPUSETS: enabled
- CONFIG_MACVLAN: enabled (as module)
- CONFIG_VETH: enabled (as module)
- CONFIG_BRIDGE: enabled (as module)
- CONFIG_BRIDGE_NETFILTER: enabled
- CONFIG_NF_NAT_IPV4: enabled (as module)
- CONFIG_IP_NF_FILTER: enabled (as module)
- CONFIG_IP_NF_TARGET_MASQUERADE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_ADDRTYPE: enabled (as module)
- CONFIG_NETFILTER_XT_MATCH_CONNTRACK: enabled (as module)
- CONFIG_NF_NAT: enabled (as module)
- CONFIG_NF_NAT_NEEDED: enabled
- CONFIG_POSIX_MQUEUE: enabled

Optional Features:
- CONFIG_MEMCG_SWAP: enabled
- CONFIG_MEMCG_SWAP_ENABLED: missing
    (note that cgroup swap accounting is not enabled in your kernel config, you can enable it by setting boot option "swapaccount=1")
- CONFIG_RESOURCE_COUNTERS: enabled
- CONFIG_BLK_CGROUP: enabled
- CONFIG_IOSCHED_CFQ: enabled
- CONFIG_CGROUP_PERF: enabled
- CONFIG_CFS_BANDWIDTH: enabled
- Storage Drivers:
  - "aufs":
    - CONFIG_AUFS_FS: enabled (as module)
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "btrfs":
    - CONFIG_BTRFS_FS: enabled (as module)
  - "devicemapper":
    - CONFIG_BLK_DEV_DM: enabled
    - CONFIG_DM_THIN_PROVISIONING: enabled (as module)
    - CONFIG_EXT4_FS: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
    - CONFIG_EXT4_FS_SECURITY: enabled
  - "overlay":
    - CONFIG_OVERLAY_FS: missing
    - CONFIG_EXT4_FS_SECURITY: enabled
    - CONFIG_EXT4_FS_POSIX_ACL: enabled
  - "zfs":
    - /dev/zfs: missing
    - zfs command: available
    - zpool command: available
```

**Comment by harche** [comment]
even with `docker run --net=host` I get the same error

**Comment by phemmer** [comment]
can you run `sudo modprobe bridge nf_nat` and provide the output. Also if there's any relevant info in `dmesg` (last few lines) that might help too.

**Comment by harche** [comment]
No output for `sudo modprobe bridge nf_nat`, 

`dmesg`
```
[188081.731347] type=1400 audit(1437668577.819:95): apparmor="STATUS" operation="profile_replace" profile="unconfined" name="docker-default" pid=25783 comm="apparmor_parser"
[188087.566635] FIREWALL: IN=wlan0 OUT= MAC=33:33:00:00:00:fb:30:3a:64:a8:59:58:86:dd SRC=fe80:0000:0000:0000:323a:64ff:fea8:5958 DST=ff02:0000:0000:0000:0000:0000:0000:00fb LEN=88 TC=0 HOPLIMIT=1 FLOWLBL=0 PROTO=UDP SPT=5353 DPT=5353 LEN=48 
[188088.590285] FIREWALL: IN=wlan0 OUT= MAC=33:33:00:00:00:fb:30:3a:64:a8:59:58:86:dd SRC=fe80:0000:0000:0000:323a:64ff:fea8:5958 DST=ff02:0000:0000:0000:0000:0000:0000:00fb LEN=88 TC=0 HOPLIMIT=1 FLOWLBL=0 PROTO=UDP SPT=5353 DPT=5353 LEN=48 
[188090.535106] FIREWALL: IN=wlan0 OUT= MAC=33:33:00:00:00:fb:30:3a:64:a8:59:58:86:dd SRC=fe80:0000:0000:0000:323a:64ff:fea8:5958 DST=ff02:0000:0000:0000:0000:0000:0000:00fb LEN=88 TC=0 HOPLIMIT=1 FLOWLBL=0 PROTO=UDP SPT=5353 DPT=5353 LEN=48 
[188099.938464] aufs au_opts_parse:1155:docker[25751]: unknown option dirperm1
[188100.045367] device vethaf35a8a entered promiscuous mode
[188100.046516] IPv6: ADDRCONF(NETDEV_UP): veth1c5f1a1: link is not ready
[188100.047734] IPv6: ADDRCONF(NETDEV_CHANGE): veth1c5f1a1: link becomes ready
[188100.113700] docker0: port 1(vethaf35a8a) entered disabled state
[188100.115344] device vethaf35a8a left promiscuous mode
[188100.115362] docker0: port 1(vethaf35a8a) entered disabled state
[188147.552249] FIREWALL: IN=wlan0 OUT= MAC=33:33:00:00:00:fb:30:3a:64:a8:59:58:86:dd SRC=fe80:0000:0000:0000:323a:64ff:fea8:5958 DST=ff02:0000:0000:0000:0000:0000:0000:00fb LEN=88 TC=0 HOPLIMIT=1 FLOWLBL=0 PROTO=UDP SPT=5353 DPT=5353 LEN=48 
[188148.575780] FIREWALL: IN=wlan0 OUT= MAC=33:33:00:00:00:fb:30:3a:64:a8:59:58:86:dd SRC=fe80:0000:0000:0000:323a:64ff:fea8:5958 DST=ff02:0000:0000:0000:0000:0000:0000:00fb LEN=88 TC=0 HOPLIMIT=1 FLOWLBL=0 PROTO=UDP SPT=5353 DPT=5353 LEN=48 
```

**Comment by phemmer** [comment]
Ahha, not related to those modules at all. The `dmesg` output shows it:

    aufs au_opts_parse:1155:docker[25751]: unknown option dirperm1

There's some mention of `dirperm` [in the docs](https://docs.docker.com/reference/builder/#known-issues-run), but I don't use AUFS, so I'm not sure if it's related.

Someone else more familiar with the aufs driver will have to chime in.

**Comment by thaJeztah** [comment]
The DirPerm1 option is something present in recent AUFS versions that works around this issue; https://github.com/docker/docker/issues/783#issuecomment-19237045. However, you should be able to run without DirPerm1 support.

(In case you missed; docker now has a new APT/YUM repository, providing dynamically linked binaries: http://blog.docker.com/2015/07/new-apt-and-yum-repos/)

**Comment by harche** [comment]
@thaJeztah I tried using dynamically linked libraries, but I hit the same issue.

**Comment by estesp** [comment]
I think this is related to libnetwork as this error is the last "real" error output before the collection of rolled up failures being from the call of `/start` on the container:
```
ERRO[0020] leaving endpoint failed: a container has already joined the endpoint
```

I asked @harche to run with debug, and I see that there is an "invalid argument" output at a low level, which is usually a (Linux kernel) syscall return code that isn't being rolled up into a clearer error condition.  I don't know if that might be related to a setns() call from libnetwork that is failing, or something else?

@harche did mention that until upgrading to Docker 1.7.1 daemon/client using the new/official apt/yum repos this installation of Docker was working fine.

**Comment by cpuguy83** [comment]
I think this is fixed on the 1.8 RC.
Care to give that a whirl?

**Comment by estesp** [comment]
Yeah, I was going to have @harche try 1.8RC or -dev, but the reason I think this is now important is that 1.7.1 is an official release and in this case happened to a working install when upgrading using the new repos (so a regression of sorts unless we can find a simple settings issue or other outside-of-Docker reason).  Additionally, I thought the `invalid argument` problem recently was an issue on master (and was related to apparmor or cgroups maybe?).  This clearly has a libnetwork namespace error in the logs, which I don't think was the case on the latest 1.8-dev master problem?

**Comment by mrjana** [comment]
@estesp I think this is a dupe of #14160. As I said in that issue, the libnetwork error that you are seeing is after the fact i.e after you get the invalid argument error. It's a spurious error. The actual error is the `invalid argument` error. We need to understand where that is coming from.

**Comment by estesp** [comment]
@mrjana hmm..does look the same.  I agree that it does appear this is coming prior to the libnetwork error--given it looks like no one has gotten to the root cause, I'll see if I can get @harche to do an `strace`  log on the daemon so we can narrow down the exact root cause/syscall which generates the invalid argument.

**Comment by harche** [comment]
@estesp @mrjana I have attached the output of strace here, https://gist.github.com/harche/12f4da18e847af150d76

**Comment by harche** [comment]
running docker without aufs, `docker -d -s vfs`  fails with same error.

**Comment by LK4D4** [comment]
@harche I think it should be `strace -f`

**Comment by estesp** [comment]
@LK4D4 I got @harche to provide me an updated strace log with the right flags.  I'm afraid there is something different during libnetwork initialization that causes the rest of the container start initialization to never begin--I will post something soon with a cut down comparison between my system (also Ubuntu 14.04.2 LTS) and @harche's.  libnetwork never performs the `setns()` call on his system, which means something related to the endpoint error causes the rest of the libnetwork initialization to be skipped (although from strace hard to map to "where"), and then the container is never started at that point.

**Comment by mrjana** [comment]
@estesp @harche Can you point me to the strace log(with all threads followed)?

**Comment by estesp** [comment]
@mrjana he sent it to me over internal transfer system, but I didn't realize the last bytes were still copying..so when I closed our chat program it deleted the file as an unfinished transfer.  Given the hour in India, I don't believe I can get it back as he isn't online, or @harche will not be around until tomorrow his time to post himself.  Sorry about that.

I was going to paste the difference between the flow from about the time the /proc/<pid>/net/ns fd is opened and the /var/run/docker/netns/<id> fd is opened and some netlink traffic happens--on my system, after that segment, I assume the netlink operations are all successful, because the next action is `setns(..)` with the fd of the libnetwork created netns, followed by the usual container start flow after that.   On his system, instead of that sequence, after some netlink traffic (which I can't parse what is happening clearly as it is dumped hex traffic over sockets), all the fds are closed followed by an unmount of the netns, and then teardown of the container (setting container metadata json status to "Running: false" and all the unmounts, cleanup, etc.).

**Comment by harche** [comment]
@mrjana @estesp I have uploaded the strace here, https://drive.google.com/file/d/0B3vI9LCU26KEOFp1ZllLdk85YTQ/view?usp=sharing

**Comment by harche** [comment]
@LK4D4 I am doing `strace -o stract-docker.log -f -v docker -d` to generate strace output

**Comment by juju2112** [comment]
This happens for me reliably when I start the docker engine with the '-b' or '-bip' argument. If I let Docker randomly assign IPs to containers, it works just fine.

**Comment by harche** [comment]
Hi, 

Another system with Ubuntu 14.04 started having the same problem today. It was working fine till yesterday.

**Comment by mrjana** [comment]
Thanks @harche for the strace logs. From the strace logs I can see that the libnetwork code most likely failed while trying to move the veth interface into the network namespace erroring out with `EINVAL`. Looking at the kernel source there isn't a lot of possibility of this particular netlink path returning `EINVAL`. Can you post the `/proc/self/mountinfo` output from your machine where the problem is happening? Also you mentioned this started happening on another Ubuntu machine? Did you start any other service in those machines? Can you stop network-manager service in your setup and reboot the machine and see if the problem is still present?

**Comment by harche** [comment]
```
 cat /proc/self/mountinfo
17 22 0:15 / /sys rw,nosuid,nodev,noexec,relatime - sysfs sysfs rw
18 22 0:3 / /proc rw,nosuid,nodev,noexec,relatime - proc proc rw
19 22 0:5 / /dev rw,relatime - devtmpfs udev rw,size=3946940k,nr_inodes=986735,mode=755
20 19 0:12 / /dev/pts rw,nosuid,noexec,relatime - devpts devpts rw,gid=5,mode=620,ptmxmode=000
21 22 0:16 / /run rw,nosuid,noexec,relatime - tmpfs tmpfs rw,size=792564k,mode=755
22 1 252:1 / / rw,relatime - ext4 /dev/mapper/ubuntu--vg-root rw,errors=remount-ro,data=ordered
24 17 0:17 / /sys/fs/cgroup rw,relatime - tmpfs none rw,size=4k,mode=755
25 17 0:18 / /sys/fs/fuse/connections rw,relatime - fusectl none rw
26 17 0:6 / /sys/kernel/debug rw,relatime - debugfs none rw
27 17 0:10 / /sys/kernel/security rw,relatime - securityfs none rw
28 21 0:19 / /run/lock rw,nosuid,nodev,noexec,relatime - tmpfs none rw,size=5120k
29 21 0:20 / /run/shm rw,nosuid,nodev,relatime - tmpfs none rw
30 21 0:21 / /run/user rw,nosuid,nodev,noexec,relatime - tmpfs none rw,size=102400k,mode=755
31 17 0:22 / /sys/fs/pstore rw,relatime - pstore none rw
32 24 0:23 / /sys/fs/cgroup/cpuset rw,relatime - cgroup cgroup rw,cpuset
33 24 0:24 / /sys/fs/cgroup/cpu rw,relatime - cgroup cgroup rw,cpu
34 24 0:25 / /sys/fs/cgroup/cpuacct rw,relatime - cgroup cgroup rw,cpuacct
35 24 0:26 / /sys/fs/cgroup/memory rw,relatime - cgroup cgroup rw,memory
36 24 0:27 / /sys/fs/cgroup/devices rw,relatime - cgroup cgroup rw,devices
37 24 0:28 / /sys/fs/cgroup/freezer rw,relatime - cgroup cgroup rw,freezer
38 24 0:29 / /sys/fs/cgroup/blkio rw,relatime - cgroup cgroup rw,blkio
39 24 0:30 / /sys/fs/cgroup/perf_event rw,relatime - cgroup cgroup rw,perf_event
40 24 0:31 / /sys/fs/cgroup/hugetlb rw,relatime - cgroup cgroup rw,hugetlb
41 22 8:1 / /boot rw,relatime - ext2 /dev/sda1 rw,stripe=4
42 21 0:32 / /run/rpc_pipefs rw,relatime - rpc_pipefs rpc_pipefs rw
43 24 0:33 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime - cgroup systemd rw,name=systemd
23 30 0:34 / /run/user/1000/gvfs rw,nosuid,nodev,relatime - fuse.gvfsd-fuse gvfsd-fuse rw,user_id=1000,group_id=1000
44 22 252:1 /var/lib/docker/aufs /var/lib/docker/aufs rw,relatime - ext4 /dev/mapper/ubuntu--vg-root rw,errors=remount-ro,data=ordered
```

**Comment by mtp401** [comment]
I have this exact issue. It seems to have started after a kernel update (3.13.0-52-generic -> 3.13.0-59-generic). If there's anything I can do to help with diagnoses let me know.

**Comment by harche** [comment]
@mtp401 I am using 3.13.0-59-generic as well.

**Comment by estesp** [comment]
can those able to reproduce the problem move up to **3.13.0-61** that is now available in the trusty repos?  I never experienced the problem but I went up to -58 the other day and now just moved to -61 and still no errors.  I wonder if it was specific to a problem in Ubuntu's -59 kernel image?

**Comment by harche** [comment]
@estesp Upgraded to 3.13.0-61, unfortunately didn't help :(  Still having the same issue.

**Comment by harche** [comment]
I booted system with 3.13.0-53, but still see the same issue.

**Comment by mtp401** [comment]
Upgraded to 3.13.0-61 and still have the same issue. Also booted to 3.13.0-52, issue still persists.

**Comment by mrjana** [comment]
@harche @mtp401 I am not able to reproduce anything either in my ubuntu 14.04. So I think at this time it boils down to some specific services or packages that you have installed on your machines. Can you provide a dump of both 
```
sudo service --status-all
```
and
```
sudo apt --installed list
```
Hope it provides some information on this behavior

**Comment by harche** [comment]
@mrjana  https://gist.github.com/harche/ab6ac38fb64c4cc7b74d

**Comment by mrjana** [comment]
@harche Is it possible for you to disable vpnagentd and ciscod and try running docker? I am suspecting that might be causing your problems. Unfortunately I can try on my end because vpn setup requires e2e anyconnect config which I cannot set it up.

**Comment by mtp401** [comment]
After a little leg work on my end this seems to be related to Symantec AV. Doing

`sudo apt-get purge savap`

on my system and rebooting fixes the issue. Reinstallting brings it back.

**Comment by andybalaam** [comment]
Confirming this is caused by Symantec AV, and sudo apt-get purge savap fixed it for me.

**Comment by andybalaam** [comment]
Note: I saw the problem on an Ubuntu 14.04 VM.  My 15.04 machine works fine, even when Symantec AV is installed.

**Comment by thaJeztah** [comment]
Interested to hear if this is indeed the cause for all people here. If it is, I think we should add a note to the [Known issues](https://github.com/docker/docker/blob/a628155a7211041b9e671980ead195ddc6e63c6b/docs/misc/release-notes.md#L58) section (which should probably be moved up). Perhaps even to the check-config.sh script?

**Comment by juju2112** [comment]
I am running RHEL 7.1, and I don't have Symantec AV installed. Here is the output of the commands mrjava requested:

https://gist.github.com/juju2112/36b7e83da84a3153c084

**Comment by juju2112** [comment]
Here is an strace:

https://gist.github.com/juju2112/80bbcbcf7394a57824fb

**Comment by juju2112** [comment]
If I remove the '-b' or '--bip' flags from the docker engine startup, the issue goes away. Is that the case for anyone else?

**Comment by aboch** [comment]
@juju2112 Given you are on RHEL kernel and based on your troubleshooting outcome, I feel you are hitting a different issue. 

It would be great if you could try the docker [1.8.0-rc1](https://test.docker.com/builds/Linux/x86_64/docker-1.8.0-rc1) binary and let us know if the problem is still there  (in case, pls attach the daemon logs).
Thanks.

**Comment by ramonvanstijn** [comment]
Removing Symantec AV is also a solution for #14160 and I think it should be
added to the Known Issues as well as to check-config.sh.

2015-08-03 12:37 GMT+02:00 Sebastiaan van Stijn <notifications@github.com>:

**Comment by stiggity** [comment]
@thaJeztah You'd asked if this was the cause for all people here. I hadn't originally been here, but I found this issue via searching my problem (the same) and the fix suggested by @mtp401 worked for me.

**Comment by thaJeztah** [comment]
@stiggity thanks for reporting that! (and glad this helped you)

**Comment by mrjana** [comment]
@mtp401 @andybalaam @stiggity Thanks for confirming that this issue is related to symantec AV. As @thaJeztah suggested we should update the known issues document to help others who are hitting this issue.

Nevertheless I feel this issue should still be open until we find out more about the interaction symantec AV and docker networking. Needless to say, it would be ideal if docker networking can work even when symantec AV service is active.

**Comment by jk-shah** [comment]
I am seeing the same problem that @juju2112  indicated on CentOS 7.0 with plus repository and using bip. (I did not try without BP) Using docker 1.5.0 did not have any issue on the same setup.

@juju2112, did you file a new tracker specific to RHEL/CentOS ?

**Comment by juju2112** [comment]
No, I did not. I have unfortunately been quite busy of late.

**Comment by flynnk** [comment]
--bip=10.2.1.0/24 fails with this error, but...
--bip=10.2.1.2/24 works perfectly fine.  This may be what is throwing folks.  The documentation here refers to the --bip as the subnet, and it's not strictly the subnet (that would be .0), but rather the IP of the first host to use.

**Comment by juju2112** [comment]
I do believe you're right. That appears to work. Thanks!

**Comment by suhastawade** [comment]
Can anyone please help me in understanding the following errors (docker logs). 
I am getting those errors when I am running "sudo docker run -it centos:centos6 /bin/bash" on my centos machine.
I am interested to know what exactly the meaning of those docker logs and when possibly they are coming and what they are doing at background which is blocked by anti-virus.
If I uninstall the anti-virus the issue goes away, but I want to know the functionality of docker.

docker info
Containers: 74
Images: 5
Storage Driver: devicemapper
 Pool Name: docker-253:0-498433-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 599.4 MB
 Data Space Total: 107.4 GB
 Data Space Available: 31.31 GB
 Metadata Space Used: 3.232 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.14.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 1
Total Memory: 1.797 GiB
Name: localhost.localdomain
ID: 7HMP:JWDF:FTXP:WUGY:GWXK:5Y5D:EECJ:RARS:MYOJ:YDWU:2NBB:QDKI
Debug mode (server): true
File Descriptors: 15
Goroutines: 22
System Time: 2015-11-20T06:39:53.633590904-05:00
EventsListeners: 0
Init SHA1: 96ca13a0ede5263a247551f6bde936069da4a609
Init Path: /usr/libexec/docker/dockerinit
Docker Root Dir: /var/lib/docker
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

docker version
Client:
 Version:      1.8.2
 API version:  1.20
 Package Version: docker-1.8.2-7.el7.centos.x86_64
 Go version:   go1.4.2
 Git commit:   bb472f0/1.8.2
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2
 API version:  1.20
 Package Version:
 Go version:   go1.4.2
 Git commit:   bb472f0/1.8.2
 Built:
 OS/Arch:      linux/amd64

docker logs
DEBU[5175] Calling POST /containers/create
INFO[5175] POST /v1.20/containers/create
DEBU[5175] [deviceset] AddDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init basehash=3bbbf0aca3593044c280551babb73f306c145fe512976015829cfc98d2b45d94)
DEBU[5175] registerDevice(153, 782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [deviceset] AddDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init basehash=3bbbf0aca3593044c280551babb73f306c145fe512976015829cfc98d2b45d94) END
DEBU[5175] activateDeviceIfNeeded(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] UnmountDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] Unmount(/var/lib/docker/devicemapper/mnt/782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] Unmount done
DEBU[5175] [devmapper] deactivateDevice(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] removeDevice START(docker-253:0-498433-782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] removeDevice END(docker-253:0-498433-782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] deactivateDevice END(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] [devmapper] UnmountDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init) END
DEBU[5175] [deviceset] AddDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f basehash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init)
DEBU[5175] registerDevice(154, 782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [deviceset] AddDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f basehash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f-init) END
DEBU[5175] activateDeviceIfNeeded(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] UnmountDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] Unmount(/var/lib/docker/devicemapper/mnt/782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] Unmount done
DEBU[5175] [devmapper] deactivateDevice(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] removeDevice START(docker-253:0-498433-782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] removeDevice END(docker-253:0-498433-782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] deactivateDevice END(782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f)
DEBU[5175] [devmapper] UnmountDevice(hash=782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f) END
DEBU[5175] Calling POST /containers/{name:.*}/attach
INFO[5175] POST /v1.20/containers/782590a3eb619e0d23f9408f1165a3cb49df4a4253b947854559f6d003c7089f/attach?stderr=1&stdin=1&stdout=1&stream=1
DEBU[5175] attach: stdin: begin
DEBU[5175] attach: stdout: begin
DEBU[5175] attach: stderr: begin
INFO[0040] POST /v1.20/containers/9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2/start
DEBU[0040] activateDeviceIfNeeded(9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
ERRO[0040] failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument
ERRO[0040] failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument
ERRO[0040] failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument
ERRO[0040] failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument
ERRO[0040] failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument
DEBU[0040] [devmapper] UnmountDevice(hash=9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] Unmount(/var/lib/docker/devicemapper/mnt/9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] Unmount done
DEBU[0040] [devmapper] deactivateDevice(9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] removeDevice START(docker-253:0-498433-9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] removeDevice END(docker-253:0-498433-9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] deactivateDevice END(9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2)
DEBU[0040] [devmapper] UnmountDevice(hash=9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2) END
ERRO[0040] Handler for POST /containers/{name:.*}/start returned error: Cannot start container 9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2: failed sandbox add: failed to add interface vethfe769a3 to sandbox: failed in prefunc: failed to set namespace on link "vethfe769a3": invalid argument
ERRO[0040] HTTP Error                                    err=Cannot start container 9f1f853f20fb50de2ef0caab4b2ab5197880323e5441b4b0c0b73553e33edeb2: failed sandbox add: failed to add interface vethfe769a3 to sandbox: failed in prefunc: failed to set namespace on link "vethfe769a3": invalid argument statusCode=500

**Comment by thaJeztah** [comment]
@suhastawade please don't leave a random, non-related question on the issue tracker; if you need help with running docker, or want to discuss using docker with other users, the #docker IRC channel, docker-users group on google or forums.docker.com are better locations for doing that.

**Comment by passion4aix** [comment]
I'm seeing the following errors on RHEL 7.1 with docker 1.9.1:

Error message captured from journalctl:

                Apr 26 17:33:56 localhost.localdomain forward-journal[1574]: time="2016-04-26T17:33:56.946991304+08:00" level=warning msg="Running modprobe bridge br_netfilter failed with message: modprobe: WARNING: Module br_netfilter not found.\ninsmod /lib/modules/3.10.0-229.14.1.el7.x86_64/kernel/net/llc/llc.ko \ninsmod /lib/modules/3.10.0-229.14.1.el7.x86_64/kernel/net/802/stp.ko \ninsmod /lib/modules/3.10.0-229.14.1.el7.x86_64/kernel/net/bridge/bridge.ko \n, error: exit status 1"
 

When starting docker service, the following errors occur:

# systemctl status docker -l
● docker.service - Docker Application Container Engine
   Loaded: loaded (/usr/lib/systemd/system/docker.service; enabled; vendor preset: disabled)
   Active: active (running) since Wed 2016-04-27 12:52:52 HKT; 5s ago
     Docs: http://docs.docker.com
Main PID: 5412 (sh)
   Memory: 14.4M
   CGroup: /system.slice/docker.service
           ├─5412 /bin/sh -c /usr/bin/docker daemon --icc=false $OPTIONS            $DOCKER_STORAGE_OPTIONS            $DOCKER_NETWORK_OPTIONS            $ADD_REGISTRY            $BLOCK_REGISTRY            $INSECURE_REGISTRY            2>&1 | /usr/bin/forward-journald -tag docker
           ├─5413 /usr/bin/docker daemon --icc=false --selinux-enabled --storage-driver devicemapper --storage-opt dm.fs=xfs --storage-opt dm.thinpooldev=/dev/mapper/docker-docker--pool --storage-opt dm.use_deferred_removal=true --storage-opt dm.use_deferred_deletion=true --add-registry registry.access.redhat.com
           └─5414 /usr/bin/forward-journald -tag docker
 
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.082929392+08:00" level=error msg="failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument"
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.087072777+08:00" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.1/16. Daemon option --bip can be used to set a preferred IP address"
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.200116003+08:00" level=error msg="failed to set to initial namespace, net:[4026531956], initns fd 9: invalid argument"
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.633533423+08:00" level=info msg="Loading containers: start."
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: ..........
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.671329858+08:00" level=info msg="Loading containers: done."
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.675346113+08:00" level=info msg="Daemon has completed initialization"
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.675870118+08:00" level=info msg="Docker daemon" commit="78ee77d/1.9.1" execdriver=native-0.2 graphdriver=devicemapper version=1.9.1
Apr 27 12:52:52 localhost.localdomain forward-journal[5414]: time="2016-04-27T12:52:52.731004400+08:00" level=info msg="API listen on /var/run/docker.sock"
Apr 27 12:52:52 localhost.localdomain systemd[1]: Started Docker Application Container Engine.

When launching a container, the following error is thrown:

# docker run hello-world
WARNING: IPv4 forwarding is disabled. Networking will not work.
Error response from daemon: Cannot start container 0eef63bd57734a430b23555db6039b014d6fa89c927b187598bdb38339077f2f: [8] System error: failed to add interface veth31eaa9a to sandbox: failed in prefunc: failed to set namespace on link "veth31eaa9a": invalid argument


Cause: Symantec Endpoint Protection for Linux 12.1.6
Workaround: UNINNSTALL Symantec Endpoint Protection for Linux 
# /opt/Symantec/symantec_antivirus/uninstall.sh and Enter Y for uninstall (default is NO)

reboot the server and everything works well...

I'll be glad to test if anyone has found a way to make exception for docker in SEP for Linux....

**Comment by trickyj** [comment]
@passion4aix 

Docker software does not run on Linux when Endpoint Protection autoprotect service is running

http://www.symantec.com/docs/TECH234544

**Solution**

Currently, SEP autoprotect must be stopped or uninstalled before Docker containers will run. Symantec is aware of this issue and this document will be updated as new information becomes available.

**Comment by uniqhemant** [comment]
Facing the same issue as harche but even on disabling the auto protect for Symantec Endpoint Protection(suggested above), I get the same error on running any docker image.

$ sudo /opt/Symantec/symantec_antivirus/sav info -a
Disabled

$ sudo docker run --net=host hello-world
invalid argument
Error response from daemon: Cannot start container b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b: [8] System error: invalid argument

Below are the details -
##########################################
$ **uname -r**
2.6.32-573.22.1.el6.x86_64

$ sudo **docker info**
Containers: 17
Images: 9
Storage Driver: devicemapper
 Pool Name: docker-253:1-804954-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 690.4 MB
 Data Space Total: 107.4 GB
 Data Space Available: 106.7 GB
 Metadata Space Used: 1.716 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.146 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.95-RHEL6 (2015-09-08)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 2.6.32-573.22.1.el6.x86_64
Operating System: <unknown>
CPUs: 4
Total Memory: 7.574 GiB
Name: oc6608656727.ibm.com
ID: FX4B:JFVE:2BTP:QEYR:AQP2:5C6X:IVR7:BDVM:3FIP:K3XQ:EOOG:W35T

**Docker logs** 

time="2016-05-12T16:03:00.987281361+05:30" level=info msg="POST /v1.19/containers/create"
time="2016-05-12T16:03:02.864682159+05:30" level=info msg="POST /v1.19/containers/5ccc1597a8d46b96964a0618243bbc82503a225d8e125bebcb120fdd8d186bd4/attach?stderr=1&stdout=1&stream=1"
time="2016-05-12T16:03:02.866109733+05:30" level=info msg="POST /v1.19/containers/5ccc1597a8d46b96964a0618243bbc82503a225d8e125bebcb120fdd8d186bd4/start"
time="2016-05-12T16:03:02.991118895+05:30" level=error msg="leaving endpoint failed: a container has already joined the endpoint"
time="2016-05-12T16:03:04.266758990+05:30" level=error msg="Handler for POST /containers/{name:.*}/start returned error: Cannot start container 5ccc1597a8d46b96964a0618243bbc82503a225d8e125bebcb120fdd8d186bd4: invalid argument"
time="2016-05-12T16:03:04.266907892+05:30" level=error msg="HTTP Error" err="Cannot start container 5ccc1597a8d46b96964a0618243bbc82503a225d8e125bebcb120fdd8d186bd4: invalid argument" statusCode=500
time="2016-05-12T16:03:34.893178656+05:30" level=info msg="POST /v1.19/containers/create"
time="2016-05-12T16:03:36.346392403+05:30" level=info msg="POST /v1.19/containers/b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b/attach?stderr=1&stdout=1&stream=1"
time="2016-05-12T16:03:36.348646998+05:30" level=info msg="POST /v1.19/containers/b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b/start"
time="2016-05-12T16:03:36.586933380+05:30" level=warning msg="exit status 1"
time="2016-05-12T16:03:37.013128307+05:30" level=error msg="Error unmounting device b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b: UnmountDevice: device not-mounted id b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b"
time="2016-05-12T16:03:37.013546068+05:30" level=error msg="Handler for POST /containers/{name:.*}/start returned error: Cannot start container b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b: [8] System error: invalid argument"
time="2016-05-12T16:03:37.013693278+05:30" level=error msg="HTTP Error" err="Cannot start container b847abcec6bfe29c2e9d1fa23012fa0b47b86124c76547be368a97ec1b85e99b: [8] System error: invalid argument" statusCode=500
time="2016-05-12T16:05:58.641487222+05:30" level=info msg="GET /v1.19/info"

**Comment by thaJeztah** [comment]
@uniqhemant RHEL 6 and kernel 2.6.x is no longer a supported platform for docker. Docker requires at least RHEL 7.x and kernel 3.10 or above

**Comment by trickyj** [comment]
I would suggest you to uninstall SEP AP (Autoprotect)  module.

**Comment by dmaziuk** [comment]
> --bip=10.2.1.0/24 fails with this error, but...
> --bip=10.2.1.2/24 works perfectly fine. This may be what is throwing folks. The documentation here refers to the --bip as the subnet, and it's not strictly the subnet (that would be .0), but rather the IP of the first host to use.

Pity there is no "like" button, I'd give you a bazillion of those... Or a "kicking myself in the head" smiley: of course a "bridge ip" should be an ip, no matter what TFM says. D'oh.

**Comment by neersighted** [comment]
Closing as very stale.
